runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java

```diff
@@ -209,6 +213,34 @@ protected void startUp() throws Exception
 		if (accountSession != null)
 		{
 			lootTrackerClient = new LootTrackerClient(accountSession.getUuid());
+
+
+			// Load all persistent data on client load if enabled
+			if (config.loadDataInClient())
+			{
+				clientThread.invokeLater(() ->
+				{
+					switch (client.getGameState())
+					{
+						case STARTING:
+						case UNKNOWN:
+							return false;
+					}
+
+					// Run blocking web requests on new thread
+					new Thread(() ->
```

Comment:
This should use executor service, and the entire thing should be limited to maybe 100-1000 loot records

Reply:
The get API route limit defaults to `1024` already so should be fine as is.

I've also updated the `new Thread()` to use `ScheduledExecutorService.submit()` instead